compiler/test/fail_compilation/diag9831.d

```diff
@@ -1,7 +1,7 @@
 /*
 TEST_OUTPUT:
 ---
-fail_compilation/diag9831.d(13): Error: function `diag9831.main.__lambda3` cannot access variable `c` in frame of function `D main`
+fail_compilation/diag9831.d(13): Error: function `diag9831.main.__lambda3(__T1)(x)` cannot access variable `c` in frame of function `D main`
```

Comment:
I am a bit worried about this because it's a bit confusing "wait, this is a template? then why does the compiler call it a function?". The explanation here comes from the fact that the type of the lambda is deduced from the lhs of the assign expression. So here we have a clash between the internal representation of the lambda and what meets the eye. I feel inclined to leave this as is, but am open to suggestions.

Reply:
It's ugly anyways how it exposes the internal identifier `__lambda`, there's room for future improvement indeed